ui/src/components/forms/Dropdown.tsx

```diff
@@ -0,0 +1,113 @@
+import { Menu, Transition } from '@headlessui/react';
+import { ChevronDownIcon } from '@heroicons/react/20/solid';
+import { Fragment } from 'react';
+import { Icon } from '~/types/Icon';
+import { classNames } from '~/utils/helpers';
+
+type DropdownAction = {
+  id: string;
+  label: string;
+  icon?: Icon;
+  onClick: () => void;
+  disabled?: boolean;
+  title?: string;
+  className?: string;
+  activeClassName?: string;
+  inActiveClassName?: string;
+};
+
+type DropdownProps = {
+  label: string;
+  actions: DropdownAction[];
+};
+
+export default function Dropdown(props: DropdownProps) {
+  const { label, actions } = props;
+  return (
+    <Menu as="div" className="relative inline-block text-left">
+      <div>
+        <Menu.Button className="inline-flex w-full justify-center gap-x-1.5 rounded-md px-3 py-2 text-sm font-semibold shadow-sm ring-2 ring-inset ring-gray-300 bg-white text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-violet-300">
+          {label}
+          <ChevronDownIcon
+            className="-mr-1 h-5 w-5 text-gray-400"
+            aria-hidden="true"
+          />
+        </Menu.Button>
+      </div>
+
+      <Transition
+        as={Fragment}
+        enter="transition ease-out duration-100"
+        enterFrom="transform opacity-0 scale-95"
+        enterTo="transform opacity-100 scale-100"
+        leave="transition ease-in duration-75"
+        leaveFrom="transform opacity-100 scale-100"
+        leaveTo="transform opacity-0 scale-95"
+      >
+        <Menu.Items className="absolute right-0 mt-2 w-56 origin-top-right divide-y divide-gray-100 rounded-md shadow-lg ring-1 ring-black ring-opacity-5 bg-white focus:outline-none">
+          {actions.map((action) => (
+            <div className="py-1" key={action.id}>
+              {!action.disabled && (
+                <Menu.Item key={action.id}>
+                  {({ active }) => (
+                    <a
+                      href="#"
+                      className={classNames(
+                        active
+                          ? action.activeClassName
+                            ? action.activeClassName
+                            : 'text-gray-900 bg-gray-100'
+                          : action.inActiveClassName
+                          ? action.inActiveClassName
+                          : 'text-gray-700',
```

Comment:
Take it or leave it (I think this is valid):
I found this a little hard to grok; I think it is equivalent (correct me if I am wrong): 
```suggestion
                        active
                          ? (action.activeClassName ?? 'text-gray-900 bg-gray-100')
                          : (action.inActiveClassName ?? 'text-gray-700'),
```